autoload/lsp/ui/vim/utils.vim

```diff
@@ -81,6 +81,16 @@ function! s:is_file_uri(uri) abort
     return stridx(a:uri, 'file:///') == 0
 endfunction
 
+if has("unix")
+    function! s:get_loc_text(path, lineno) abort
+        " Use sed to avoid loading the entire file into memory.
+        return system("sed -n " . a:lineno . "p < " . shellescape(a:path))
```

Comment:
this is sync, so it will block vim.

Reply:
Ah yes, I've had success with this on a pretty large project, but I can see how it could be bad on a slow network filesystem, or if there's no limit to the number of files. I'll take a stab at making this async (and behind a config flag) hopefully sometime in the next few weeks.